src/device-manager.ts

```diff
@@ -0,0 +1,160 @@
+export interface AudioDevice {
+  deviceId: string;
+  label: string;
+  default: boolean;
+  current: boolean;
+}
+
+export interface DeviceManagerCallbacks {
+  onDeviceError: (error: Error) => void;
+  onDeviceDisconnected?: (deviceId: string) => void;
+  onDeviceSwitched?: (fromDeviceId: string, toDeviceId: string) => void;
+}
+
+export interface DeviceManagerOptions {
+  /** Enable automatic device switching when current device is disconnected */
+  autoSwitchOnDisconnect?: boolean;
+  /** Enable device change event listening */
+  listenForDeviceChanges?: boolean;
+}
+
+interface MediaDeviceWithDefault extends MediaDeviceInfo {
+  default?: boolean;
+}
+
+export class DeviceManager {
+  private wavRecorder: any; // WavRecorder instance
+  private callbacks: DeviceManagerCallbacks;
+  private currentDeviceId: string | null = null;
+  private options: Required<DeviceManagerOptions>;
+  private deviceChangeListener: ((devices: MediaDeviceWithDefault[]) => void) | null = null;
+
+  constructor(wavRecorder: any, callbacks: DeviceManagerCallbacks, options: DeviceManagerOptions = {}) {
+    this.wavRecorder = wavRecorder;
+    this.callbacks = callbacks;
+    this.options = {
+      autoSwitchOnDisconnect: options.autoSwitchOnDisconnect ?? true,
+      listenForDeviceChanges: options.listenForDeviceChanges ?? true,
+    };
+
+    this.updateCurrentDeviceId();
+
+    if (this.options.listenForDeviceChanges) {
```

Comment:
When would we want to not listen? Can we remove the listenForDeviceChanges option?